src/llfs/testing/test_config.cpp

```diff
@@ -0,0 +1,87 @@
+//#=##=##=#==#=#==#===#+==#+==========+==+=+=+=+=+=++=+++=+++++=-++++=-+++++++++++
+//
+// Part of the LLFS Project, under Apache License v2.0.
+// See https://www.apache.org/licenses/LICENSE-2.0 for license information.
+// SPDX short identifier: Apache-2.0
+//
+//+++++++++++-+-+--+----- --- -- -  -  -   -
+
+#include <llfs/testing/test_config.hpp>
+//
+
+#include <batteries/env.hpp>
+
+namespace llfs {
+namespace testing {
+
+//==#==========+==+=+=++=+++++++++++-+-+--+----- --- -- -  -  -   -
+//
+TestConfig::TestConfig() noexcept
+{
+  this->extra_testing_ =                          //
+      batt::getenv_as<int>("LLFS_EXTRA_TESTING")  //
+          .value_or(0);
+
+  this->low_level_log_device_sim_ =                          //
+      batt::getenv_as<int>("LLFS_LOW_LEVEL_LOG_DEVICE_SIM")  //
+          .value_or(1);
+}
+
+//==#==========+==+=+=++=+++++++++++-+-+--+----- --- -- -  -  -   -
+//
+bool TestConfig::extra_testing() const noexcept
+{
+  return this->extra_testing_;
+}
+
+//==#==========+==+=+=++=+++++++++++-+-+--+----- --- -- -  -  -   -
+//
+bool TestConfig::low_level_log_device_sim() const noexcept
+{
+  return this->low_level_log_device_sim_;
+}
+
+//==#==========+==+=+=++=+++++++++++-+-+--+----- --- -- -  -  -   -
+//
+const std::filesystem::path& TestConfig::project_dir() noexcept
+{
+  if (this->project_dir_.empty()) {
+    this->project_dir_ = []() {
+      const char* project_dir_env = std::getenv("PROJECT_DIR");
+      if (project_dir_env) {
+        return std::filesystem::path{project_dir_env};
+      }
+
+      std::filesystem::path file_path{__FILE__};
+      if (std::filesystem::exists(file_path)) {
```

Comment:
Since we start from `__FILE__`, yes we have a specific expectation about the directory structure being navigated.  This is expressed by the names of the `std::filesystem::path` variables.  

Yes, we could make it a loop, but I'm not convinced this is better.  What do you think the best argument in favor of loop-izing this code is?

Reply:
Done!